File: module-server/src/main/java/de/twomartens/wahlrecht/model/internal/VotingResult.java
package de.twomartens.wahlrecht.model.internal;

import java.util.Map;
import lombok.Builder;
import lombok.Getter;

@Getter
public class VotingResult {

  private final NominationId nominationId;
  private final int votesOnNomination;
  private final int votesThroughHealing;
  private final Map<Integer, Integer> votesPerPosition;
  private Integer totalVotes = null;

  @Builder
  public VotingResult(NominationId nominationId, int votesOnNomination, int votesThroughHealing,
      Map<Integer, Integer> votesPerPosition) {
    this.votesOnNomination = votesOnNomination;
    this.votesThroughHealing = votesThroughHealing;
    this.votesPerPosition = votesPerPosition;
    this.nominationId = nominationId;
  }

  public int getTotalVotes() {
    if (totalVotes == null) {
      totalVotes = votesOnNomination + votesThroughHealing + votesPerPosition.values().stream()
          .reduce(0, Integer::sum);
    }
    return totalVotes;
  }

  public int getTotalVotesWithoutHealing() {
    return getTotalVotes() - votesThroughHealing;
  }
}

File: module-server/src/main/kotlin/de/twomartens/wahlrecht/model/internal/VotingResult.kt
package de.twomartens.wahlrecht.model.internal

data class VotingResult(
    val nominationId: NominationId, val votesOnNomination: Int,
    val votesThroughHealing: Int,
    val votesPerPosition: Map<Int, Int>
) {
    val totalVotes = votesOnNomination + votesThroughHealing + votesPerPosition.values
            .reduce{ acc: Int, value: Int -> acc + value }
    val totalVotesWithoutHealing = totalVotes - votesThroughHealing
}
